app/controllers/draws_controller.rb

```diff
@@ -222,4 +207,34 @@ def prepare_students_edit_data
     @available_students_count = UngroupedStudentsQuery.call.where(draw_id: nil)
                                                       .count
   end
+
+  def process_bulk_assignment
+    result = DrawStudentsUpdate.update(draw: @draw,
+                                       params: students_update_params)
+    @students_update = result[:update_object]
+    if @students_update
+      prepare_students_edit_data
+      result[:action] = 'student_summary'
+    else
+      result[:path] = student_summary_draw_path(@draw)
+    end
+    result
```

Comment:
That is an excellent point, but since this was just a simple move I think we should add that to Refactoring.